setup.py

```diff
@@ -102,6 +101,7 @@
         "lazyasd>=0.1.4",
         "packaging>=23.0,<24",
         "pandas>=1.3.0,<2",
+        "pip",
```

Comment:
I would rather fail early if we detect `pip` is not in your PATH, since we are not using pip as a library but only via `subprocess.call`. This way, `pip` does no have to ship with Ape.

Reply:
`pip` or `uv pip` detected, then we can proceed (since `uv pip` is cli-compatible with `pip`)

Reply:
okay so we could make `uv pip` stand-in fallback for `pip`....